check_tools/find_bad_code_snippets.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -eo pipefail
+
+mkdir -p BUILD
+echo '*' > BUILD/.mbedignore
+
+# simple fake main for snippets without a main function
+cat > fake_main.c <<MAIN
+#include "mbed_toolchain.h"
+MBED_WEAK int main(void) { return 0; }
+MAIN
+
+function compile {
+    if [ "$1" != "NOCI" -a "$1" != "TODO" ]
+    then
+        PYTHONPATH=mbed-os python mbed-os/tools/make.py -t GCC_ARM -m ${1:-K64F} --source=. --build=BUILD/${1:-K64F}/GCC_ARM | sed '/\(error\|warning\)/I!d'
+    fi
+}
+     
+for f in $(find -name mbed-os -prune -o -name '*.md' -print)
+do
+    # compile c++ snippets
+    for l in $(sed -n '/``` *\(cpp\|c++\)\($\| \+\)/I=' $f)
+    do
+        echo "Validating cpp $f:$l ..."
+        echo "#include \"mbed.h\"" > main.cpp
```

Comment:
Haha, I thought we were suppose to _not_ be doing this.

Reply:
Well the examples assume mbed is present don't they?

Reply:
I think for the purpose of just seeing if they compile its fine and much better than adding a bunch of manual include paths to individual snippets.